Framework/DataObjects/src/BasePeak.cpp

```diff
@@ -0,0 +1,477 @@
+  return getInitialEnergy() - getFinalEnergy();
+}
+
+//----------------------------------------------------------------------------------------------
+/** Get the H index of the peak */
+double BasePeak::getH() const { return m_H; }
+
+/** Get the K index of the peak */
+double BasePeak::getK() const { return m_K; }
+
+/** Get the L index of the peak */
+double BasePeak::getL() const { return m_L; }
+
+/** Return the HKL vector */
+Mantid::Kernel::V3D BasePeak::getHKL() const { return V3D(m_H, m_K, m_L); }
+
+/** Return True if the peak has been indexed */
+bool BasePeak::isIndexed() const {
+  if (m_H == 0. && m_K == 0. && m_L == 0.)
+    return false;
+  return true;
+}
+
+/** Return the int HKL vector */
+Mantid::Kernel::V3D BasePeak::getIntHKL() const { return m_intHKL; }
+
+/** Return the int MNP vector */
+V3D BasePeak::getIntMNP() const { return m_intMNP; }
+
+//----------------------------------------------------------------------------------------------
+/** Set the H index of this peak
+ * @param m_H :: index to set   */
+void BasePeak::setH(double m_H) { this->m_H = m_H; }
+
+/** Set the K index of this peak
+ * @param m_K :: index to set   */
+void BasePeak::setK(double m_K) { this->m_K = m_K; }
+
+/** Set the L index of this peak
+ * @param m_L :: index to set   */
+void BasePeak::setL(double m_L) { this->m_L = m_L; }
+
+/** Set the BankName of this peak
+ * @param m_bankName :: index to set   */
+void BasePeak::setBankName(std::string m_bankName) {
+  this->m_bankName = std::move(m_bankName);
+}
+
+/** Set all three H,K,L indices of the peak */
+void BasePeak::setHKL(double H, double K, double L) {
+  m_H = H;
+  m_K = K;
+  m_L = L;
+}
+
+/** Set all HKL
+ *
+ * @param HKL :: vector with x,y,z -> h,k,l
+ */
+void BasePeak::setHKL(const Mantid::Kernel::V3D &HKL) {
+  m_H = HKL.X();
+  m_K = HKL.Y();
+  m_L = HKL.Z();
+}
+
+/** Set int HKL
+ *
+ * @param HKL :: vector with integer x,y,z -> h,k,l
+ */
+void BasePeak::setIntHKL(const V3D &HKL) {
+  m_intHKL = V3D(std::round(HKL[0]), std::round(HKL[1]), std::round(HKL[2]));
+}
+
+/** Sets the modulated peak structure number
+ * @param MNP :: modulated peak structure value
+ */
+void BasePeak::setIntMNP(const V3D &MNP) {
+  m_intMNP = V3D(std::round(MNP[0]), std::round(MNP[1]), std::round(MNP[2]));
+}
+
+//----------------------------------------------------------------------------------------------
+/** Return the # of counts in the bin at its peak*/
+double BasePeak::getBinCount() const { return m_binCount; }
+
+/** Return the integrated peak intensity */
+double BasePeak::getIntensity() const { return m_intensity; }
+
+/** Return the error on the integrated peak intensity */
+double BasePeak::getSigmaIntensity() const { return m_sigmaIntensity; }
+
+/** Return the peak intensity divided by the error in the intensity */
+double BasePeak::getIntensityOverSigma() const {
+  const auto result = m_intensity / m_sigmaIntensity;
+  return (std::isinf(result)) ? 0.0 : result;
```

Comment:
Should we trap nan as well?
```suggestion
  return (std::isfinite(result)) ? result : 0.0;
```